source/SettingsWindow.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2020, thaflo, Ilio Catallo, Stefano Celentano, Eleonora Ciceri, all rights reserved
+ * Distribuited under the terms of the GPL v2 license
+ *
+ * Authors:
+ *
+ *			thaflo
+ *
+ * December 2020: thaflo: Change color directly (that's the Be way)
+ */
+
+#include "SettingsWindow.h"
+#include <ColorControl.h>
+#include <stdio.h>
+
+// Libraries
+#include <Alert.h>
+#include <Button.h>
+#include <private/shared/CalendarView.h>
+
+// Messages
+#define COLOR_CHANGED 	'ccrq'
+#define REVERT_COLOR		'revc'
+#define DEFAULT_COLOR		'defc'
+#define SETTINGS_CLOSE		'_sec'
+#define COLOR_CHOSEN		'_cch'
+#define SAVE		'save'
+
+//translation
+#include <Catalog.h>
+#include <TranslationUtils.h>
+
+#undef B_TRANSLATION_CONTEXT
+#define B_TRANSLATION_CONTEXT "SettingsWindow"
+
+using namespace BPrivate;
+
+/*
+* Constructor
+* It is created with the dimensions of BRect
+*/
+SettingsWindow :: SettingsWindow (BRect frame, BHandler *handler, rgb_color color)
+			: BWindow (frame, B_TRANSLATE("Note settings"), B_TITLED_WINDOW, B_NOT_RESIZABLE) {
+
+	// Variables
+	BPoint 				 leftTop(10.0, 50.0);
+	color_control_layout matrix;
+	long 				 cellSide;
+
+	BButton				 *revertButton;
+	BButton				*setDefButton;
+
+	// Create the  view and set the background color, then add child to window
+	frame.OffsetTo(B_ORIGIN);
+	fColorView = new ColorView (frame, "ColorView",handler);
+	fColorView->SetViewColor(ui_color(B_PANEL_BACKGROUND_COLOR));
+	fColorView->ResizeToPreferred();
+	AddChild(fColorView);
+
+	// Drawing...
+	fMessenger = new BMessenger(handler);
+
+	matrix = B_CELLS_32x8;
+	cellSide = 9.0;
+
+	fColorControl = new BColorControl (leftTop, matrix, cellSide, "ColorControl", new BMessage(COLOR_CHANGED));
+	fColorControl -> SetValue(color);
+	fColorView -> AddChild(fColorControl);
+
+	revertButton = new BButton (BRect(10, 150, 170, 175), "default", B_TRANSLATE("Default"), new BMessage(REVERT_COLOR));
```

Comment:
```suggestion
	revertButton = new BButton (BRect(10, 150, 170, 175), "default", B_TRANSLATE("Defaults"), new BMessage(REVERT_COLOR));
```

Normally, it's "Defaults", plural, in prefs panels